src/coredump/stacktrace.c

```diff
@@ -113,40 +117,209 @@ static int thread_callback(Dwfl_Thread *thread, void *userdata) {
+                                        return DWARF_CB_ABORT;
+                                }
+                                r = json_variant_merge(c->package_metadata, w);
+                                if (r < 0) {
+                                        log_error_errno(r, "json_variant_merge of package meta with buildid failed: %m");
+                                        return DWARF_CB_ABORT;
+                                }
+
+                                /* Finally stash the name, so we avoid double visits. */
+                                name_key = strdup(name);
+                                if (!name_key) {
+                                        log_oom();
+                                        return DWARF_CB_ABORT;
+                                }
+                                r = set_ensure_consume(c->modules, &string_hash_ops, name_key);
+                                if (r < 0) {
+                                        log_error_errno(r, "set_ensure_consume failed: %m");
+                                        return DWARF_CB_ABORT;
+                                }
+
+                                return DWARF_CB_OK;
+                        }
+                }
+        }
+
+        /* Didn't find package metadata for this module - that's ok, just go to the next. */
+        return DWARF_CB_OK;
+}
+
+static int module_callback(Dwfl_Module *mod, void **userdata, const char *name, Dwarf_Addr start, void *arg) {
+        _cleanup_(json_variant_unrefp) JsonVariant *id_json = NULL;
         struct stack_context *c = arg;
-        GElf_Addr id_vaddr;
+        size_t n_program_headers;
+        GElf_Addr id_vaddr, bias;
         const unsigned char *id;
-        int id_len;
+        int id_len, r;
+        Elf *elf;
 
         assert(mod);
         assert(c);
 
         if (!name)
                 name = "(unnamed)"; /* For logging purposes */
 
-        fprintf(c->f, "Found module %s", name);
-
         /* We are iterating on each "module", which is what dwfl calls ELF objects contained in the
          * core file, and extracting the build-id first and then the package metadata.
          * We proceed in a best-effort fashion - not all ELF objects might contain both or either.
          * The build-id is easy, as libdwfl parses it during the dwfl_core_file_report() call and
          * stores it separately in an internal library struct. */
         id_len = dwfl_module_build_id(mod, &id, &id_vaddr);
-        if (id_len == 0) {
-                fprintf(c->f, " without build-id\n");
-                return DWARF_CB_OK;
+        if (id_len <= 0) {
+                /* If we don't find a build-id, note it in the journal message, and try
+                 * anyway to find the package metadata. It's unlikely to have the latter
+                 * without the former, but there's no hard rule. */
+                fprintf(c->f, "Found module %s without build-id\n", name);
+        } else {
```

Comment:
no {} around single-line if blocks.

maybe add full stop to end of sentence in the fprintf() message